modules/repository/license.go

```diff
@@ -6,13 +6,89 @@ package repository
 import (
 	"bufio"
 	"bytes"
+	"context"
 	"fmt"
 	"regexp"
 	"strings"
 
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/container"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/options"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
+
+	licenseclassifier "github.com/google/licenseclassifier/v2"
+)
+
+var (
+	classifier   *licenseclassifier.Classifier
+	sameLicenses map[string]string
 )
 
+func init() {
+	err := loadSameLicenses()
+	if err != nil {
+		log.Error("loadSameLicenses: %v", err)
+	}
+	err = initClassifier()
+	if err != nil {
+		log.Error("initClassifier: %v", err)
+	}
+}
+
+func loadSameLicenses() error {
+	data, err := options.AssetFS().ReadFile("", "sameLicenses")
```

Comment:
Is `camelCase` a good naming style for the file? IIRC the option assets seldom used such naming style.

Reply:
How about `SameLicenses.json`?